crashreports/internal/win32/CrashReportContext.cpp

```diff
@@ -0,0 +1,115 @@
+#include <locale>
+#include <codecvt>
+#include <sstream>
+#include "client/windows/handler/exception_handler.h"
+#include "CrashReportContext.h"
+
+namespace {
+	bool uploadReport(
+		const wchar_t* dump_path,
+		const wchar_t* minidump_id,
+		void* context,
+		EXCEPTION_POINTERS* exinfo,
+		MDRawAssertionInfo* assertion,
+		bool succeeded)
+	{
+		CrashReportContext* crashReportContext = reinterpret_cast<CrashReportContext*>(context);
+		if (!crashReportContext->send(dump_path, minidump_id))
+			return false;
+		return succeeded;
+	}
+
+	std::string stringifyParameters(const std::map<std::string, std::string>& parameters)
+	{
+		std::stringstream stream;
+
+		std::size_t parameterIndex = 0;
+		std::size_t parametersCount = parameters.size();
+		for (auto& pair : parameters)
+		{
+			stream << pair.first.c_str() << "=\\\"" << pair.second.c_str() << "\\\"";
+			++parameterIndex;
+			if (parameterIndex < parametersCount)
+				stream << ",";
+		}
+		return stream.str();
+	}
+}
+
+bool CrashReportContext::setSenderPathUTF8(const std::string& path)
+{
+    return wcscpy_s(senderPath, MaxBufferLength, std::wstring_convert<std::codecvt_utf8<wchar_t>, wchar_t>().from_bytes(path).c_str()) == 0;
```

Comment:
wcscpy_s is not necessitated by crash context here.  Is there another reason to prefer it to string class operations?

Reply:
lots of things could go wrong, but keeping this data in a static storage should make it less vulnerable i guess

Reply:
Yes, maybe that is a good point.  Even though this function is not one of those called after the crash.